pkg/gcptarget/gcptarget.go

```diff
@@ -250,30 +268,57 @@ func checkPathGlob(expression string) error {
 func checkPathGlobs(rs []string) error {
 	for idx, r := range rs {
 		if err := checkPathGlob(r); err != nil {
-			return errors.Wrapf(err, "idx: %d", idx)
+			return fmt.Errorf("idx [%d]: %w", idx, err)
 		}
 	}
 	return nil
 }
 
 // ValidateConstraint implements client.TargetHandler
 func (g *GCPTarget) ValidateConstraint(constraint *unstructured.Unstructured) error {
-	targets, found, err := unstructured.NestedStringSlice(constraint.Object, "spec", "match", "target")
-	if err != nil {
-		return errors.Errorf("invalid spec.match.target: %s", err)
-	}
-	if found {
-		if err := checkPathGlobs(targets); err != nil {
-			return errors.Wrapf(err, "invalid glob in target")
+	ancestries, ancestriesFound, ancestriesErr := unstructured.NestedStringSlice(constraint.Object, "spec", "match", "ancestries")
+	targets, targetsFound, targetsErr := unstructured.NestedStringSlice(constraint.Object, "spec", "match", "target")
+	if ancestriesFound && targetsFound {
+		return errors.New("only one of spec.match.ancestries and spec.match.target can be specified")
+	} else if ancestriesFound {
+		if ancestriesErr != nil {
+			return fmt.Errorf("invalid spec.match.ancestries: %s", ancestriesErr)
+		}
+		if ancestriesErr := checkPathGlobs(ancestries); ancestriesErr != nil {
+			return fmt.Errorf("invalid glob in spec.match.ancestries: %w", ancestriesErr)
+		}
+	} else if targetsFound {
+		glog.Warning(
+			"spec.match.target is deprecated and will be removed in a future release. Use spec.match.ancestries instead",
```

Comment:
This isn't quite released yet - but yes, once it's released, it would be great if policy-library could be updated to use `ancestries` and `excludedAncestries`

Reply:
Thanks.